src/main/java/ee/tuleva/onboarding/auth/jwt/JwtAuthorizationFilter.java

```diff
@@ -36,30 +39,42 @@ protected void doFilterInternal(
       }
       try {
         if (SecurityContextHolder.getContext().getAuthentication() == null) {
-          AuthenticatedPerson principal =
-              principalService.getFrom(
-                  jwtTokenUtil.getPersonFromToken(jwtToken),
-                  jwtTokenUtil.getAttributesFromToken(jwtToken));
+          TokenType tokenType = jwtTokenUtil.getTypeFromToken(jwtToken);
+          if (tokenType == TokenType.ACCESS) {
+            AuthenticatedPerson principal =
+                principalService.getFrom(
+                    jwtTokenUtil.getPersonFromToken(jwtToken),
+                    jwtTokenUtil.getAttributesFromToken(jwtToken));
 
-          final var authorities =
-              jwtTokenUtil.getAuthoritiesFromToken(jwtToken).stream()
-                  .map(SimpleGrantedAuthority::new)
-                  .toList();
+            final var authorities =
+                jwtTokenUtil.getAuthoritiesFromToken(jwtToken).stream()
+                    .map(SimpleGrantedAuthority::new)
+                    .toList();
 
-          final var authenticationToken =
-              new UsernamePasswordAuthenticationToken(principal, jwtToken, authorities);
+            final var authenticationToken =
+                new UsernamePasswordAuthenticationToken(principal, jwtToken, authorities);
 
-          authenticationToken.setDetails(
-              new WebAuthenticationDetailsSource().buildDetails(request));
+            authenticationToken.setDetails(
+                new WebAuthenticationDetailsSource().buildDetails(request));
 
-          SecurityContextHolder.getContext().setAuthentication(authenticationToken);
+            SecurityContextHolder.getContext().setAuthentication(authenticationToken);
+          }
         }
       } catch (ExpiredJwtException e) {
         logger.info("JWT Token is expired");
+        respondWithTokenExpired(response);
+        return;
```

Comment:
Is it OK to return in a Security Filter? shouldn't it continue with the whole filter chain?

Reply:
Ok, from the [docs](https://docs.oracle.com/javaee/6/api/javax/servlet/Filter.html#doFilter%28javax.servlet.ServletRequest,%20javax.servlet.ServletResponse,%20javax.servlet.FilterChain%29) I see that:
- **Either** invoke the next entity in the chain using the FilterChain object (chain.doFilter()),
- **or** not pass on the request/response pair to the next entity in the filter chain to block the request processing